presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -504,13 +509,50 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                     newJoinFilter.isPresent() == node.getFilter().isPresent() &&
                             (!newJoinFilter.isPresent() || areExpressionsEquivalent(newJoinFilter.get(), node.getFilter().get()));
 
+            Map<String, Symbol> dynamicFilters = ImmutableMap.of();
+            List<Expression> predicates = ImmutableList.of();
+            if (node.getType() == INNER && isEnableDynamicFiltering(session)) {
+                // New equiJoinClauses could potentially not contain symbols used in current dynamic filters.
+                // Since we use PredicatePushdown to push dynamic filters themselves,
+                // instead of separate ApplyDynamicFilters rule we derive dynamic filters within PredicatePushdown itself.
+                // Even if equiJoinClauses match, current dynamic filters may not match equiJoinClauses
+                Set<Symbol> equiJoinClauseSymbols = equiJoinClauses.stream()
+                        .map(JoinNode.EquiJoinClause::getRight)
+                        .collect(toImmutableSet());
+                ImmutableMap.Builder<String, Symbol> dynamicFiltersBuilder = ImmutableMap.builder();
+                node.getDynamicFilters().entrySet().stream()
```

Comment:
You can simply create new dynamic filters from scratch here, no need to look at previous filters.